libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -477,6 +477,73 @@ bool AP_Mission::set_current_cmd(uint16_t index)
     return true;
 }
 
+// returns false on any issue at all.
+bool AP_Mission::set_item(uint16_t index, mavlink_mission_item_int_t& src_packet)
+{
+        // this is the on-storage format
+        AP_Mission::Mission_Command cmd;
+
+        // can't handle request for anything bigger than the mission size+1...
+        if (index > num_commands() ) {  return false; }
+
+        // convert from mavlink-ish format to storage format, if we can.
+        if (mavlink_int_to_mission_cmd(src_packet, cmd) != MAV_MISSION_ACCEPTED) {
+            return false;
+        }
+
+        // A request to set the 'next' item after the end is how we add an extra
+        //  item to the list, thus allowing us to write entire missions if needed.
+        if (index == num_commands() ) {  return add_cmd( cmd); }
+
+        // replacing an existing mission item...
+        return AP_Mission::replace_cmd( index, cmd);
+}
+
+bool AP_Mission::get_item(uint16_t index, mavlink_mission_item_int_t& ret_packet)
+{
+        // setting ret_packet.command = -1  and/or returning false 
+        //  means it contains invalid data after it leaves here.
+
+        // this is the on-storage format
+        AP_Mission::Mission_Command cmd;
+
+        // can't handle request for anything bigger than the mission size...
+        if (index >= num_commands() ) { ret_packet.command = -1;  return false; }
+
+        mavlink_mission_request_int_t packet{}; // minimal placeholder
```

Comment:
why is packet var needed at all?

Reply:
agreed, this var is no longer required, fixing it now.